script.js

```diff
@@ -1,4 +1,275 @@
+  if (hasPlayer1Rolled == 1 && hasPlayer2Rolled == 1) {
+    return `Player 2, you chose ${capitalizer(
+      userInput
+    )} first. <br>Your number is ${player2ConcatenatedNumber}. <br>It is now time to see who won! Type in "compare" to see who won.`;
+  }
+};
+
+//Create a function that compares the two numbers and output a who wins statement.
+var comparingWinnerStatement = function () {
+  //Create a variable that is a statement to let the players know how to reset the game
+  var howToReset = `Please input "reset" if you wish to reset the game and play again.`;
+
+  //Check if there is a tie
+  if (player1ConcatenatedNumber == player2ConcatenatedNumber) {
+    return `Both of your numbers are ${player1ConcatenatedNumber}. <br><br>It's a tie! <br><br>${howToReset}`;
+  }
+
+  //Check if player 1 won
+  if (player1ConcatenatedNumber > player2ConcatenatedNumber) {
+    return `Player 1's number is ${player1ConcatenatedNumber}. <br>Player 2's number is ${player2ConcatenatedNumber}. <br><br>Player 1 won! <br><br>${howToReset}`;
+  }
+
+  //Check if player 2 won
+  if (player2ConcatenatedNumber > player1ConcatenatedNumber) {
+    return `Player 1's number is ${player1ConcatenatedNumber}. <br>Player 2's number is ${player2ConcatenatedNumber}. <br><br>Player 2 won! <br><br>${howToReset}`;
+  }
+};
+
+//Create a function that resets the game
+var gameReset = function () {
+  hasPlayer1Rolled = 0;
+  hasPlayer2Rolled = 0;
+  player1DiceRolls = [];
+  player2DiceRolls = [];
+  player1ConcatenatedNumber = 0;
+  player2ConcatenatedNumber = 0;
+  timeToReset = 0;
+  return `The game has been reset! You may play again by pressing the "Submit" button.`;
+};
+
+//Create a function that receives the user input and returns the correct output statements
+var beatThatGame = function (userInput) {
+  //Make the input entirely lowercase
+  userInput = userInput.toLowerCase();
+
+  if (
+    !(userInput == "dice 1" || userInput == "dice 2") &&
+    ((hasPlayer1Rolled == 1 && player1ConcatenatedNumber == 0) ||
+      (hasPlayer1Rolled == 1 &&
+        hasPlayer2Rolled &&
+        player2ConcatenatedNumber == 0))
+  ) {
+    return;
+  }
+  if (userInput == "" && !(hasPlayer1Rolled == 1 && hasPlayer2Rolled == 1)) {
+    //Check if there is no input, which means the players are rolling the dice. Call the dice rolling functions
+    rollTheDice();
+    return statementForDiceRolled();
+  }
+
+  //Check if input is "dice 1" or "dice 2", which means the players are choosing which dice is the first number and which is the second. Call the concatenating functions
+  if (userInput == "dice 1" || userInput == "dice 2") {
+    concatenator(userInput);
+    return statementForConcatenation(userInput);
+  }
+
+  //Check if input is "compare", which means its time to compare the two numbers and see who won
+  if (
+    userInput == "compare" &&
+    hasPlayer1Rolled == 1 &&
+    hasPlayer2Rolled == 1
+  ) {
+    timeToReset = 1;
+    return comparingWinnerStatement();
+  }
```

Comment:
again, probably not the best from a user experience point of view for the user to have to type 'compare' in order to get the outcome of the game. if both players have already rolled their dice and selected their order, then the only next logical progression for the game is to compare the scores. would be better if you just ignore whatever input is given and immediately move on to your `comparingWinnerStatement` function.